test/test.sh

```
@@ -105,151 +167,225 @@ assert() {
+	assert 'cat | cat | ls\n\n'
+	echo
+	
+	# Expand Variable
+	printf "# Expand Variable\n"
+	assert 'echo $USER'
+	assert 'echo $USER$PATH&TERM'
+	assert 'echo "$USER $PATH $TERM"'
+	echo
+	
+	# Special Parameter $?
+	printf "# Special Parameter $?\n"
```

Shellcheck (0.11.0)

[warning] This $? refers to echo/printf, not a previous command. Assign to variable to avoid it being overwritten.

(SC2320)